--- glib-2.50.2/configure.ac	2016-11-07 17:31:00.000000000 +0200
+++ glib-2.50.2_patched/configure.ac	2016-12-01 13:14:20.000000000 +0200
@@ -63,7 +63,7 @@
 # Save this value here, since automake will set cflags later
 cflags_set=${CFLAGS:+set}
 
-AM_INIT_AUTOMAKE([1.11 -Wno-portability no-define no-dist-gzip dist-xz tar-ustar])
+AM_INIT_AUTOMAKE([1.11 -Wno-portability no-define no-dist-gzip dist-xz tar-ustar subdir-objects])
 AM_MAINTAINER_MODE([enable])
 
 # Support silent build rules. Disable
