diff -rupN minidlna/inotify.c minidlnanew/inotify.c
--- minidlna/inotify.c	2013-11-15 23:25:08.206588313 +0000
+++ minidlnanew/inotify.c	2013-11-15 22:19:53.000000000 +0000
@@ -336,12 +336,14 @@ inotify_insert_file(char * name, const c
 				return -1;
 			break;
 		case TYPE_VIDEO:
-			if( !is_video(path) )
+			if( !is_video(path) &&
+			    !is_playlist(path) )
 				return -1;
 			break;
 		case TYPE_VIDEO|TYPE_IMAGES:
 			if( !is_image(path) &&
-			    !is_video(path) )
+			    !is_video(path) &&
+			    !is_playlist(path) )
 				return -1;
 			break;
 		case TYPE_IMAGES:
diff -rupN minidlna/scanner.c minidlnanew/scanner.c
--- minidlna/scanner.c	2013-11-15 23:25:08.226589087 +0000
+++ minidlnanew/scanner.c	2013-11-15 21:54:11.000000000 +0000
@@ -649,8 +650,9 @@ filter_v(scan_filter *d)
 {
 	return ( filter_hidden(d) &&
 	         (filter_type(d) ||
-		  (d->d_type == DT_REG &&
-	           is_video(d->d_name)))
+		  ((d->d_type == DT_REG) &&
+		   (is_video(d->d_name) ||
+	            is_playlist(d->d_name))))
 	       );
 }
 
@@ -661,7 +663,8 @@ filter_vp(scan_filter *d)
 	         (filter_type(d) ||
 		  ((d->d_type == DT_REG) &&
 		   (is_video(d->d_name) ||
-	            is_image(d->d_name))))
+		    is_image(d->d_name) ||
+	            is_playlist(d->d_name))))
 	       );
 }
 
diff -rupN minidlna/utils.c minidlnanew/utils.c
--- minidlna/utils.c	2013-11-15 23:25:08.266588993 +0000
+++ minidlnanew/utils.c	2013-11-15 22:32:10.000000000 +0000
@@ -475,7 +475,8 @@ resolve_unknown_type(const char * path,
 						type = TYPE_FILE;
 					break;
 				case TYPE_VIDEO:
-					if( is_video(path) )
+					if( is_video(path) ||
+					    is_playlist(path) )
 						type = TYPE_FILE;
 					break;
 				case TYPE_IMAGES:
