diff --git a/configure.ac b/configure.ac
index 6e9e8e8..eec5236 100644
--- a/configure.ac
+++ b/configure.ac
@@ -85,6 +85,7 @@ dnl ==========================================================================
 # Checks for programs.
 AC_PROG_CC
 AC_PROG_INSTALL
+PKG_PROG_PKG_CONFIG
 AC_MSG_CHECKING([for native Win32])
   case $build_os in
     mingw*) os_win32=yes ;;
diff --git a/lib/Makefile.am b/lib/Makefile.am
index 2e113da..f96bc6d 100644
--- a/lib/Makefile.am
+++ b/lib/Makefile.am
@@ -4,12 +4,14 @@ lib_LTLIBRARIES = libfribidi.la
 
 AM_CPPFLAGS = $(MISC_CFLAGS)
 
-libfribidi_la_LDFLAGS = -no-undefined -version-info $(LT_VERSION_INFO) -export-symbols-regex "^fribidi_.*"
+libfribidi_la_LDFLAGS = -no-undefined -version-info $(LT_VERSION_INFO)
 libfribidi_la_LIBADD = $(MISC_LIBS)
 libfribidi_la_DEPENDENCIES =
 
 if OS_WIN32
 libfribidi_la_LDFLAGS += -export-symbols $(srcdir)/fribidi.def
+else
+libfribidi_la_LDFLAGS += -export-symbols-regex "^fribidi_.*"
 endif # OS_WIN32
 
 if FRIBIDI_CHARSETS
diff --git a/lib/common.h b/lib/common.h
index 55a01ee..3e484b0 100644
--- a/lib/common.h
+++ b/lib/common.h
@@ -45,10 +45,6 @@
 # define FRIBIDI_PRIVATESPACE(SYMBOL) FRIBIDI_PRIVATESPACE0(_,FRIBIDI_NAMESPACE(_##SYMBOL##__internal__))
 #endif /* !FRIBIDI_PRIVATESPACE */
 
-#if (defined(WIN32)) || (defined(_WIN32_WCE))
-# define FRIBIDI_ENTRY __declspec(dllexport)
-#endif /* WIN32 */
-
 #if FRIBIDI_USE_GLIB+0
 # ifndef SIZEOF_LONG
 #  define SIZEOF_LONG GLIB_SIZEOF_LONG
diff --git a/lib/fribidi-bidi.c b/lib/fribidi-bidi.c
index 42a5fab..89d6546 100644
--- a/lib/fribidi-bidi.c
+++ b/lib/fribidi-bidi.c
@@ -463,6 +463,13 @@ fribidi_get_par_embedding_levels (
 	  move_node_before (pp, explicits_list);
 	  pp = &temp_link;
 	}
+      else if (this_type == FRIBIDI_TYPE_BS)
+	{
+	  /* X8. All explicit directional embeddings and overrides are
+	     completely terminated at the end of each paragraph. Paragraph
+	     separators are not included in the embedding. */
+	  break;
+	}
       else
 	{
 	  /* X6. For all types besides RLE, LRE, RLO, LRO, and PDF:
@@ -475,11 +482,6 @@ fribidi_get_par_embedding_levels (
 	  if (!FRIBIDI_IS_NEUTRAL (override))
 	    RL_TYPE (pp) = override;
 	}
-      /* X8. All explicit directional embeddings and overrides are
-         completely terminated at the end of each paragraph. Paragraph
-         separators are not included in the embedding. */
-      /* This function is running on a single paragraph, so we can do
-         X8 after all the input is processed. */
     }
 
     /* Implementing X8. It has no effect on a single paragraph! */
@@ -546,6 +548,10 @@ fribidi_get_par_embedding_levels (
 	    pp = merge_with_prev (pp);
 	  else
 	    RL_TYPE (pp) = prev_type;
+	  if (prev_type == next_type && RL_LEVEL (pp) == RL_LEVEL (pp->next))
+	    {
+	      pp = merge_with_prev (pp->next);
+	    }
 	  continue;		/* As we know the next condition cannot be true. */
 	}
 
@@ -564,7 +570,7 @@ fribidi_get_par_embedding_levels (
 
     last_strong = base_dir;
     /* Resolving dependency of loops for rules W4 and W5, W5 may
-       want to prevent W4 to take effect in the next turn, do this 
+       want to prevent W4 to take effect in the next turn, do this
        through "w4". */
     w4 = true;
     /* Resolving dependency of loops for rules W4 and W5 with W7,
diff --git a/lib/fribidi-bidi.h b/lib/fribidi-bidi.h
index 1b463c6..dd2de67 100644
--- a/lib/fribidi-bidi.h
+++ b/lib/fribidi-bidi.h
@@ -43,7 +43,7 @@
 
 #include "fribidi-begindecls.h"
 
-#define fribidi_get_par_embedding_levels FRIBIDI_NAMESPACE(get_par_embedding_levels)
+#define fribidi_get_par_direction FRIBIDI_NAMESPACE(get_par_direction)
 /* fribidi_get_par_direction - get base paragraph direction
  *
  * This function finds the base direction of a single paragraph,
diff --git a/lib/fribidi-common.h b/lib/fribidi-common.h
index b8f0d07..6dede39 100644
--- a/lib/fribidi-common.h
+++ b/lib/fribidi-common.h
@@ -53,11 +53,7 @@
 
 /* FRIBIDI_ENTRY is a macro used to declare library entry points. */
 #ifndef FRIBIDI_ENTRY
-# if (defined(WIN32)) || (defined(_WIN32_WCE))
-#  define FRIBIDI_ENTRY __declspec(dllimport)
-# else /* !WIN32 */
 #  define FRIBIDI_ENTRY		/* empty */
-# endif	/* !WIN32 */
 #endif /* !FRIBIDI_ENTRY */
 
 #if FRIBIDI_USE_GLIB+0
diff --git a/test/Makefile.am b/test/Makefile.am
index 8859f22..ad63fd9 100644
--- a/test/Makefile.am
+++ b/test/Makefile.am
@@ -1,5 +1,5 @@
 
-TESTS_ENVIRONMENT = /bin/sh
+TESTS_ENVIRONMENT=srcdir=${srcdir} top_builddir=${top_builddir} EXEEXT=${EXEEXT} sh
 
 TESTS = run.tests
 
diff --git a/test/run.tests b/test/run.tests
index 295c4bd..65113dc 100755
--- a/test/run.tests
+++ b/test/run.tests
@@ -36,7 +36,7 @@ if test -z "$VERBOSE"; then
    exec > /dev/null 2>&1
 fi
 
-if ! test -f "$top_builddir/bin/fribidi"; then
+if ! test -f "$top_builddir/bin/fribidi$EXEEXT"; then
   echo "run.tests: you must make fribidi first"
   exit 1
 fi
@@ -48,11 +48,11 @@ TEST () {
   charset="${testcase#*_}"
   charset="${charset%%_*}"
   echo -n "=== $test === "
-  if ! "$top_builddir/bin/fribidi" --charset "$charset" </dev/null >/dev/null 2>&1; then
+  if ! "$top_builddir/bin/fribidi$EXEEXT" --charset "$charset" </dev/null >/dev/null 2>&1; then
     echo " [Character set not supported]"
     return 0
   fi
-  "$top_builddir/bin/fribidi" --test --charset "$charset" "$testcase" > "$test.output"
+  "$top_builddir/bin/fribidi$EXEEXT" --test --charset "$charset" "$testcase" > "$test.output"
 
   reference="${testcase%.input}.reference";
   test -f "$reference" || reference="tests/${reference##*/}"
diff --git a/test/test_CapRTL_implicit.input b/test/test_CapRTL_implicit.input
index cd5fadf..ef6a3d5 100644
--- a/test/test_CapRTL_implicit.input
+++ b/test/test_CapRTL_implicit.input
@@ -24,3 +24,4 @@ A TEST FOR WEAK TYPES: 123+,456
 ANOTHER FOR WEAK TYPES: A123,456
 MORE WEAK TYPES: hooloo123,456
 and TEST|L1 ||too
+_<+`+1
diff --git a/test/test_CapRTL_implicit.reference b/test/test_CapRTL_implicit.reference
index 6ef244a..e324a7e 100644
--- a/test/test_CapRTL_implicit.reference
+++ b/test/test_CapRTL_implicit.reference
@@ -24,3 +24,4 @@ A TEST FOR WEAK TYPES: 123+,456     =>     456,123+ :SEPYT KAEW ROF TSET A
 ANOTHER FOR WEAK TYPES: A123,456    =>    123,456A :SEPYT KAEW ROF REHTONA
 MORE WEAK TYPES: hooloo123,456      =>      hooloo123,456 :SEPYT KAEW EROM
 and TEST|L1 ||too                   => and TSET|1L ||too
+_<+`+1                              =>                                +`+1
